src/nlsolve/newton.jl

```diff
@@ -70,12 +68,15 @@ Equations II, Springer Series in Computational Mathematics. ISBN
 
     # compute norm of residuals
     iter > 1 && (ndzprev = ndz)
-    ndz = integrator.opts.internalnorm(dz, tstep)
+    atmp = calculate_residuals(dz, uprev, u, integrator.opts.abstol, integrator.opts.reltol,integrator.opts.internalnorm, t)
```

Comment:
why is this supposed to be scaled by tolerance?

Reply:
We want to make sure that `dz`'s error is small enough to not cause step rejection.

Reply:
I see, smart.